internal/ast/ast.go

```diff
@@ -9113,93 +9105,46 @@ func (node *JSDocTemplateTag) Clone(f NodeFactoryCoercible) *Node {
 
 func (node *JSDocTemplateTag) TypeParameters() *TypeParameterList { return node.typeParameters }
 
-// JSDocPropertyTag
-type JSDocPropertyTag struct {
+// JSDocParameterTag, which includes both @parameter and @property
+type JSDocParameterTag struct {
 	JSDocTagBase
 	name           *EntityName
 	IsBracketed    bool
 	TypeExpression *TypeNode
 	IsNameFirst    bool
 }
 
-func (f *NodeFactory) NewJSDocPropertyTag(tagName *IdentifierNode, name *EntityName, isBracketed bool, typeExpression *TypeNode, isNameFirst bool, comment *NodeList) *Node {
-	data := &JSDocPropertyTag{}
+func (f *NodeFactory) newJSDocParameterOrPropertyTag(kind Kind, tagName *IdentifierNode, name *EntityName, isBracketed bool, typeExpression *TypeNode, isNameFirst bool, comment *NodeList) *Node {
+	data := &JSDocParameterTag{}
```

Comment:
I feel like we should probably rename the base struct type here to `JSDocParameterOrPropertyTag` and alias `type JSDocParameterTag = JSDocParameterOrPropertyTag` - likewise, the `As` cast function should change to refer to the `Or` type, but it should be aliased to the subtypes for per-use clarity. As-is, something like
```
switch (node.Kind) {
  case ast.KindJSDocParamterTag, ast.KindJSDocPropertyTag:
    node.AsJSDocParameterTag() /// ???
    
}
```
just looks super sus, even though it (currently) works because of the underlying shapes - the aliased types and methods should help with that. Should also help if we ever get around to paying off the massive debt all this type erasure is incurring and start statically checking unions again...

Reply:
sigh...you're probably right. I hate that `@param` and `@property` are the same thing except that they appear in mutually exclusive contexts. Feels bad to make them a single kind like true synonyms of jsdoc, `@augments` and `@extends` for example.